praw/helpers.py

```diff
@@ -111,6 +112,88 @@ def valid_redditors(redditors, sub):
             if resp['ok']]
 
 
+def all_submissions(reddit_session, subreddit, highest_timestamp=None, lowest_timestamp=0, verbosity=1):
+    """Yield submissions between ``highest_timestamp`` and ``lowest_timestamp``
+
+    If both ``highest_timestamp`` and ``lowest_timestamp`` are unspecified,
+    yields all submissions in the ``subreddit``.
+
+    Submissions are yielded from newest to oldest(like in the "new" queue).
+
+    :param reddit_session: The reddit_session to make requests from. In all the
+        examples this is assigned to the variable ``r``.
+    :param subreddit: Either a subreddit object, or the name of a
+        subreddit. Use `all` to get the submissions stream for all submissions
+        made to reddit.
+    :param highest_timestamp: The upper bound for ``created_utc`` attribute
+        of submissions. (Default: current unix time)
+    :param lowest_timestamp: The lower bound for ``created_utc`` atributed of
+        submissions. (Default: 0).
+        NOTE: both highest_timestamp and lowest_timestamp are proper
+        unix timestamps(just like ``created_utc`` attributes)
+    :param verbosity: A number that controls the amount of output produced to
+        stderr. <= 0: no output; >= 1: output the total number of submissions
+        processed; >= 2: output debugging information regarding
+        the search queries. (Default: 1)
+
+    """
+    def debug(msg, level):
+        if verbosity >= level:
+            sys.stderr.write(msg + '\n')
+
+    if highest_timestamp is None:
+        # convert normal unix timestamps into broken reddit timestamps
+        highest_timestamp = int(time.time()) + 28800
```

Comment:
Slightly confused: can you clarify what you mean by 'broken' and, if it is indeed broken, you may want to file an issue on reddit/reddit